torch/accelerator/_utils.py

```diff
@@ -1,10 +1,10 @@
 from typing import Optional
 
 import torch
-from torch.types import Device as _device_t
```

Comment:
This pattern is done on purpose so that it doesn't look like we're re-exposing a `torch.accelerator.Device`.
Especially because this is only a type and not an actual device object constructor.

I think the overall change in this PR to move everything to re-use `torch.types.Device` is great, I would keep the as _device_t though to not re-expose this type everywhere.